src/library/azureVirtualMachine.ts

```diff
@@ -17,7 +17,7 @@ export interface IVirtualMachineInfo {
 export class AzureVirtualMachine {
     private _TAG = "AzureVirtualMachine";
 
-    public getAzureComputeMetadata(config: Config, callback: (vm: IVirtualMachineInfo) => void) {
+    public async getAzureComputeMetadata(config: Config): Promise<IVirtualMachineInfo> {
```

Comment:
I don't think you need to tag as async because your not using await

Reply:
Changed to use await as well